### DISCUSSION TITLE
my graph doesnt show up in shiny app

### DISCUSSION THREAD

**Discussion by mchen899** [teamalgoritma/community#705] [Q&A] [answered]
server
      world_max <- 
        world %>%  
        filter(Population_category %in% "select_input")%>% 
        ggplot(aes(x=fct_reorder(Country,Population_2023), y = Population_2023, text = label))+ 
        geom_bar(stat='Identity') +   
        labs(x="Country", y = "Population in 2023", title="Top 10 Most Populated Countries 2023")+
        coord_flip()+ 
        geom_col(aes(fill = Population_2023)) +
        scale_fill_gradient(low = "grey", high = "navy") +
        theme(legend.position="none") +
        theme(plot.title = element_text(hjust = 0.5)) +
        theme(axis.text.x = element_blank()) +
        theme(axis.title.x = element_blank())
      
      # Plot Interaktif
      ggplotly(world_max, tooltip = "text")

UI

                #Untuk select input
                fluidRow(
                  box(width = 12,
                      radioButtons(inputId = "select_input",
                                  label = "Choose Category",
                                  choices = c("TOP","BELOW")

Output

![image](https://github.com/teamalgoritma/community/assets/134918404/c12adbad-9a85-4736-b4d2-c08413fd61f9)

Whats wrong with the code? Please help

**Comment by ramdhinafinita** [comment]
Dear pak @mchen899 

Pada input choices = c("TOP", "BUTTON") ini apakah nilai ini ada pada kolom Population_category? Jika ada, sebaiknya bisa dengan `unique(world$Population_category)` karena isi dari kategori inilah yang akan dijadikan filtering di server.R pada line 
```
world %>%
filter(Population_category %in% "input$select_input")
```
Jadi alurnya: Bapak membuat ui untuk pilihan dari kategori di kolom Population_category yang mana nanti isi radiobutton akan muncul sebanyak kategorinya. 
Jika sudah muncul kategorinya, bapak melakukan filtering dinamis pada server sesuai inputan user. 

Semoga membantu.

**Reply by mchen899** [reply]
masih belum bisa ke muncul graphnya, saya coba masukan input$select_input didalam title kok munculnya kolom yang lain ya?
kira2 apa yang salah?

**Reply by ramdhinafinita** [reply]
Betul pak, akan berbeda karena yang dimunculkan yakni sesuai dengan isi dari `choices`. Sebelumnya saya ingin konfirmasi, yang ingin ditampilkan di plotnya ini berdasarkan kategori apa? karena seharusnya kategori yang digunakan sama dengan filter() yang ada di server. Mungkin bisa di cek terlebih dahulu `unique(Population_category)` terlebih dahulu.

**Reply by mchen899** [reply]
berdasarkan kategori kolom Population_category.
server
      # Plot statis
      world_max <- 
        world %>%  
        filter(Population_category %in% "input$select_input")%>% 
        ggplot(aes(x=fct_reorder(Country,Population_2023), y = Population_2023, text = label))+ 
        geom_bar(stat='Identity') +   
        labs(x="Country", y = "Population in 2023", title= glue("Top 10 {input$select_input} Populated Countries"))+
        coord_flip()+ 
        geom_col(aes(fill = Population_2023)) +
        scale_fill_gradient(low = "grey", high = "navy") +
        theme(legend.position="none") +
        theme(plot.title = element_text(hjust = 0.5)) +
        theme(axis.text.x = element_blank()) +
        theme(axis.title.x = element_blank())
      
      # Plot Interaktif
      ggplotly(world_max, tooltip = "text")
      

    })

UI
                #Untuk select input
                fluidRow(
                  box(width = 12,
                      radioButtons(inputId = "select_input",
                                  label = "Choose Category",
                                  choices = c("TOP","BELOW")
                                  ))
                  
                ),

```{r}
unique(world$Population_category)
```
[1] "MIDDLE" "TOP"    "BELOW" 

seperti ini bu

**Comment by ramdhinafinita** [comment, marked as answer]
Bisa dikirimkan juga mungkin pak folder.zip projectnya beserta datasetnya, agar kami bisa cek

**Reply by mchen899** [reply]
https://github.com/teamalgoritma/community/issues/706

**Comment by mchen899** [comment]
@ramdhinafinita bu saya mau menambahkan geom densty ke geom line saya. akan tetapi selalu error. mengapa ya bu?
